fairscale/optim/adascale.py

```diff
@@ -387,6 +387,44 @@ def _update_avg(self, name: str, value: np.ndarray, factor: float) -> None:
             else:
                 self._state[name] = factor * self._state[name] + (1.0 - factor) * value
 
+    def _gather_flat_grad(self) -> torch.Tensor:
+        """
+        Helper function for gathering all gradients into a single vector.
+        Duplicated from torch.optim.lbfgs.
+        """
+        views = []
+        for param_group in self._optimizer.param_groups:
```

Comment:
perhaps rewrite this in a list comprehension way? Like:

[fn(x) for x in param_group["params"] for param_group in self._opt...]

in python, list comprehension is much faster than for loops. You can see some test I had here:

https://github.com/min-xu-ai/py_perf/blob/master/list_flatten/py_test.py

Reply:
I am just a perf improvement. no logical changes in the code